### PR TITLE
Navigate to homescreen only via go router

### PR DESCRIPTION
**Pull request by Daniela510**
## Summary by CodeRabbit

- **New Features**
	- Updated navigation logic in the Summary Screen and Leave Game Dialog to use named routes for improved user experience.
  
- **Bug Fixes**
	- Streamlined transitions to the profile selection page, enhancing navigation efficiency.

### CODE REVIEW

File: lib/features/family/features/reflect/presentation/pages/summary_screen.dart [hunk 1]
```diff
@@ -79,13 +81,13 @@ class _SummaryScreenState extends State<SummaryScreen> {
...
               ),
+              //hmm
+
               const Spacer(),
               FunButton(
                 onTap: () {
-                  Navigator.of(context).popUntil(
-                    ModalRoute.withName(
-                      FamilyPages.profileSelection.name,
-                    ),
```
Comment (MaikelStuivenberg) [inline lib/features/family/features/reflect/presentation/pages/summary_screen.dart:88]: Previous solution was clearing the stack first, is that not needed anymore?

Comment (MaikelStuivenberg) [reply lib/features/family/features/reflect/presentation/pages/summary_screen.dart:88]: Somewhere else in the app we also have a clear solution for this navigation (with a for loop I think)

Comment (MaikelStuivenberg) [reply lib/features/family/features/reflect/presentation/pages/summary_screen.dart:88]: Ah okay! Maybe that was with a push what I have seen :)

Did a search on Google: With go() "Navigating to a destination in GoRouter will replace the current stack of screens with the screens configured to be displayed for the destination route."

Comment (MaikelStuivenberg) [reply lib/features/family/features/reflect/presentation/pages/summary_screen.dart:88]: I guess that is the same as named :)

**Review by MaikelStuivenberg — APPROVED** [review]

File: lib/features/family/features/reflect/presentation/pages/summary_screen.dart (hunk 1 above)
Comment (Daniela510) [reply lib/features/family/features/reflect/presentation/pages/summary_screen.dart:88]: > Navigating to a destination in GoRouter will replace the current stack of screens with the screens configured to be displayed for the destination route. 
[GoRouter docs](https://pub.dev/documentation/go_router/latest/topics/Navigation-topic.html)

**Pull request merged**